cachi2/interface/cli.py

```diff
@@ -408,16 +428,33 @@ def merge_sboms(
     first to produce SBOMs to merge.
     """
     sboms_to_merge = []
+    if sbom_type == SBOMFormat.cyclonedx:
+        sbom_cls: Union[Type[Sbom], Type[SPDXSbom]] = Sbom
+    else:
+        sbom_cls = SPDXSbom
     for sbom_file in sbom_files_to_merge:
         try:
-            sboms_to_merge.append(Sbom.model_validate_json(sbom_file.read_text()))
-        except pydantic.ValidationError:
+            sboms_to_merge.append(sbom_cls.model_validate_json(sbom_file.read_text()))
+        except pydantic.ValidationError as e:
+            print(e)
             raise UnexpectedFormat(f"{sbom_file} does not appear to be a valid Cachi2 SBOM.")
-    sbom = Sbom(
-        components=merge_component_properties(
-            chain.from_iterable(s.components for s in sboms_to_merge)
+    if sbom_type == SBOMFormat.cyclonedx:
+        sbom: Union[Sbom, SPDXSbom] = Sbom(
+            components=merge_component_properties(
+                chain.from_iterable(cast(Sbom, s).components for s in sboms_to_merge)
+            )
         )
-    )
+    else:
+        packages = chain.from_iterable(cast(SPDXSbom, s).packages for s in sboms_to_merge)
```

Comment:
What is the reason for adding casts here and below?

Reply:
because mypy by default inherited type Union[Sbom, SPDXSbom]

Reply:
I played a bit with the code and could not come up with a less ugly way to placate mypy. I guess we'll have to continue defacing the code to keep it happy. I would also like to highlight the officially suggested way of handling similar issues: https://mypy.readthedocs.io/en/stable/error_code_list.html#check-that-attribute-exists-in-each-union-item-union-attr. It is just incredible.

Reply:
I have just realized that restructuring this code can solve the issue. You'll need two properly typed helper functions for generating actual merged SBOMs.